pypdf/_text_extraction/_text_extractor.py

```diff
@@ -357,3 +236,165 @@ def _handle_tj(
         actual_str_size["str_widths"] += font_widths
 
         return text, rtl_dir, actual_str_size
+
+    def _flush_text(self) -> None:
+        """Flush accumulated text to output and call visitor if present."""
+        self.output += self.text
+        if self.visitor_text is not None:
+            self.visitor_text(self.text, self.memo_cm, self.memo_tm, self.cmap[3], self.font_size)
+        self.text = ""
+        self.memo_cm = self.cm_matrix.copy()
+        self.memo_tm = self.tm_matrix.copy()
+
+    # Operation handlers
+
+    def _handle_bt(self, operands: List[Any]) -> None:
+        """Handle BT (Begin Text) operation - Table 5.4 page 405."""
+        self.tm_matrix = [1.0, 0.0, 0.0, 1.0, 0.0, 0.0]
+        self._flush_text()
+
+    def _handle_et(self, operands: List[Any]) -> None:
+        """Handle ET (End Text) operation - Table 5.4 page 405."""
+        self._flush_text()
+
+    def _handle_q(self, operands: List[Any]) -> None:
+        """Handle q (Save graphics state) operation - Table 4.7 page 219."""
+        self.cm_stack.append(
+            (
+                self.cm_matrix,
+                self.cmap,
+                self.font_size,
+                self.char_scale,
+                self.space_scale,
+                self._space_width,
+                self.TL,
+            )
+        )
+
+    def _handle_Q(self, operands: List[Any]) -> None:
```

Comment:
I would appreciate if we could adhere to PEP 8 naming conventions here as well, id est only using lowercase characters in the method name in this specific case.

Reply:
There was _handle_q and _handle_Q, so I could not simply make it lowercase.

I now renamed them to _handle_save_graphics_state and _handle_restore_graphics_state.

Should I name the others also by what they do instead of the operator?

Reply:
I have no strong opinion on the other cases at the moment as it is an internal API anyway. My alternative would have been to use `_handle_q` and `_handle_q_upper`, but this is just a matter of taste.

Just indicate if you consider this PR complete from your side for me to have a final look at it and merge it.

Reply:
I think it's ready :-)